### PR TITLE
QQE-737 | Make Kafka use KRaft mode

### PR DESCRIPTION
**Pull request by fedinskiy**
### Summary

In KRaft mode Zookeper is not used, since brokers communicate to each other directly.
Requires a new release of the TF

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [x] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by fedinskiy** [comment]
run tests

File: messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java [hunk 1]
```diff
@@ -48,6 +49,14 @@ public class KafkaGratefulShutdownIT {
 
     @Order(1)
     @Test
+    public void testConnection() throws InterruptedException {
+        Thread.sleep(5000); // TODO:  https://github.com/quarkusio/quarkus/issues/41441
```
Comment (michalvavrik) [inline messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java:53]: Waiting 5 s is bit of a blow. Please at the very least adjust comment based on the https://github.com/quarkusio/quarkus/issues/41441 feedback.

**Review by michalvavrik — APPROVED** [review]
lgtm, one tiny request

File: messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java (hunk 1 above)
Comment (michalvavrik) [reply messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java:53]: hmm, if you plan to do this in the future and just want to keep TODO, ok.

**Review by michalvavrik — COMMENTED** [review]
OCP failures looks related:

io.quarkus.ts.messaging.kafka.producer.OpenShiftKafkaSnappyIT.checkCompressCodecSnappy
io.quarkus.ts.messaging.kafka.reactive.streams.OperatorOpenShiftAmqStreamsKafkaStreamIT.(?)

File: messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java (hunk 1 above)
Comment (fedinskiy) [reply messaging/kafka-streams-reactive-messaging/src/test/java/io/quarkus/ts/messaging/kafka/reactive/streams/KafkaGratefulShutdownIT.java]: Add topic creation and lowered it to 1 second

**Comment by michalvavrik** [comment]
run tests

**Comment by mjurc** [comment]
This also needs to be backported to the other supported branches too, since this also enables the now supported usage of Kafka in KRaft mode without deprecated Debezium.

**Pull request merged**